### PR TITLE
feat: [SRM-11355]: Dynatrace template

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

Screenshot 2022-08-11 at 10 15 22 AM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 20 Issues 
* Code issue 20 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10492)

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx [hunk 1]
```diff
@@ -96,42 +138,88 @@ export default function DynatraceMetricPacksToService(props: DynatraceMetricPack
...
+                name={'dynatraceService'}
+                placeholder={
+                  servicesListLoading
+                    ? getString('loading')
+                    : getString('cv.healthSource.connectors.Dynatrace.servicePlaceholder')
                 }
-                onRetry={() => onValidate(metricValues.serviceMethods || [], metricValues.metricData)}
+                items={dynatraceServiceOptions}
+                label={getString('cv.healthSource.connectors.Dynatrace.servicesLabel')}
+                {...getInputGroupProps(() =>
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx:200]: props getter pattern is nice. But here it is not a right use case for that. It adds complexity to the code here.

it's purpose is if we want to add some additional functionalities on top of it's usual behaviour. 

Example:
You have your own shared state in a hook or component and you want to update that state on click of any elements, we can use it. 

Here we can simply give those props.

Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx:200]: Need to discuss this later.

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx [hunk 2]
```diff
@@ -59,10 +84,26 @@ export default function DynatraceMetricPacksToService(props: DynatraceMetricPack
...
     return mapServiceListToOptions(servicesListData?.data || [])
   }, [servicesListData?.data])
 
+  useEffect(() => {
+    if (!isConnectorRuntimeOrExpression) {
+      refetchServiceList()
+    }
+  }, [isConnectorRuntimeOrExpression])
+
+  React.useEffect(() => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx:98]: remove `React.` here

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx [hunk 3]
```diff
@@ -59,10 +84,26 @@ export default function DynatraceMetricPacksToService(props: DynatraceMetricPack
...
+    if (
+      getTypeOfInput(connectorIdentifier) !== MultiTypeInputType.FIXED &&
+      getTypeOfInput(metricDataSelectedService as string) !== MultiTypeInputType.FIXED
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx:101]:
```ts
[getTypeOfInput(connectorIdentifier) , getTypeOfInput(metricDataSelectedService as string)].every()
```

Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx:101]: current check has lesser number of comparisons

File: src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/Dynatrace/components/DynatraceMetricPacksToService/DynatraceMetricPacksToService.tsx]: done

**Review by praneshg239 — APPROVED** [review]

**Review by vasuki-kalluri — APPROVED** [review]

**Pull request merged**